trillian/ctfe/serialize_test.go

```diff
@@ -151,3 +142,35 @@ func TestSignV1SCTForPrecertificate(t *testing.T) {
 		t.Fatalf("TBS cert mismatch, got %v, expected %v", got, want)
 	}
 }
+
+func TestSignV1TreeHead(t *testing.T) {
+	signer, err := setupSigner(fakeSignature)
+	if err != nil {
+		t.Fatalf("could not create signer: %v", err)
+	}
+
+	sth := ct.SignedTreeHead{
+		Version:   ct.V1,
+		TreeSize:  10,
+		Timestamp: 1512993312000,
+	}
+	if err := signV1TreeHead(signer, &sth); err != nil {
+		t.Errorf("signV1TreeHead()=%v; want nil", err)
+	}
+	if diff := pretty.Compare(sth.TreeHeadSignature.Signature, fakeSignature); diff != "" {
```

Comment:
Quick drive by - maybe this would be better split out as an explicit cache test?

Reply:
I thought about that but the cache test then starts with exactly the same contents as the bare `TestSignV1TreeHead` test.